sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -46,28 +47,42 @@ internal BearerTokenChallengeAuthenticationPolicy(TokenCredential credential, IE
         }
 
         /// <inheritdoc />
-        public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
         {
-            return ProcessAsync(message, pipeline, true);
+            await AuthenticateRequestAsync(message, true).ConfigureAwait(false);
```

Comment:
nit: Could this be called in the private implementation of `ProcessAsync` rather than here and below in `Process`? It would make it so all the process logic is consolidated in one method.